v2/api/network/customizations/load_balancer_extension.go

```diff
@@ -120,33 +113,137 @@ func fuzzySetInboundNatRules(lb genruntime.ARMResourceSpec, inboundNatRules []ge
 
 	inboundNatRulesField := propertiesValue.FieldByName("InboundNatRules")
 	if (inboundNatRulesField == reflect.Value{}) {
-		return errors.Errorf("couldn't find InboundNatRules field on LoadBalancer")
+		return reflect.Value{}, errors.Errorf("couldn't find InboundNatRules field on LoadBalancer")
 	}
 
 	if inboundNatRulesField.Type().Kind() != reflect.Slice {
-		return errors.Errorf("inboundNatRules field on LoadBalancer was not of kind Slice")
+		return reflect.Value{}, errors.Errorf("InboundNatRules field on LoadBalancer was not of kind Slice")
+	}
+
+	return inboundNatRulesField, nil
+}
+
+func getNameField(natValue reflect.Value) (ret reflect.Value, err error) {
+	defer func() {
+		if x := recover(); x != nil {
+			err = errors.Errorf("caught panic: %s", x)
+		}
+	}()
+
+	nameField := natValue.FieldByName("Name")
+	if (nameField == reflect.Value{}) {
+		return reflect.Value{}, errors.Errorf("couldn't find name field")
+	}
+
+	nameField = reflect.Indirect(nameField)
+
+	return nameField, nil
+}
+
+func getRawInboundNatRulesField(lb map[string]any) ([]any, error) {
+	props, ok := lb["properties"]
+	if !ok {
+		return nil, errors.Errorf("couldn't find properties field on LoadBalancer")
+	}
+
+	propsMap, ok := props.(map[string]any)
+	if !ok {
+		return nil, errors.Errorf("properties field wasn't a map")
+	}
+
+	inboundNatRules, ok := propsMap["inboundNatRules"]
+	if !ok {
+		return nil, errors.Errorf("couldn't find inboundNatRules field on LoadBalancer")
+	}
+
+	inboundNatRulesSlice, ok := inboundNatRules.([]any)
+	if !ok {
+		return nil, errors.Errorf("inboundNatRules field wasn't a slice")
+	}
+
+	return inboundNatRulesSlice, nil
+}
```

Comment:
Can we avoid the repetition here? What about defining a `dynamicObject` as an alias for `map[string]any` as a container for some helper methods?

Reply:
Refactored things to share a bunch more code.

LoadBalancer is a bit different because it needs to merge rather than overwrite, and that got more complicated with this refactor because now we're looking at Azure state which means that we need to deal with conflicts consistently (previously we merged state from LB in ASO and Routes in ASO, now we're merging LB from ASO w/ LB from Azure, and since LB from Azure will have the routes defined on LB in ASO we have to make sure we choose the ASO ones). So specifically for loadbalancer it doesn't use all of the shared logic it has a bit of its own for that, but the rest of the examples have significantly reduced duplication and even LB makes use of a good bit of that.